web/src/features/time/TimeController.tsx

```diff
@@ -39,7 +39,7 @@ export default function TimeController({ className }: { className?: string }) {
         index: datetimes.length - 1,
       });
     }
-  }, [data]);
+  }, [data, datetimes, setSelectedDatetime]);
```

Comment:
Why is this added?

Reply:
datetimes.length is called inside the hook which requires datetimes to be a part of the dependency array, not sure why setSelectedDatetime is required but it would not be happy without it.